src/spectral/utils/spectral-validator.js

```diff
@@ -57,12 +59,22 @@ const setup = async function(spectral) {
   spectral.registerFormat('oas3', isOpenApiv3);
 
   // load the spectral ruleset, either a user's or the default ruleset
-  const spectralRuleset = await config.getSpectralRuleset(
-    defaultSpectralRuleset
+  const spectralRulesetURI = await config.getSpectralRuleset(
+    defaultSpectralRulesetURI
   );
 
+  // Combine user ruleset with the default ruleset
+  // The default ruleset will take precendence in any rules it defines (including extended rules)
+  await spectral.loadRuleset([defaultSpectralRulesetURI, spectralRulesetURI]);
```

Comment:
So this handles the case in which users define their own configuration with the Spectral format? But we don't let them override our default Spectral rules with that format?

I think this is what we want but as someone not very familiar with Spectral, I find it a little confusing. I appreciate the comments here, they are helpful, but will you add a couple more to provide context for why this is there?

Reply:
Matthew correct me if I'm wrong but with this PR this is what the user can do now:
1. They can override the severity level and disable/enable our spectral rules from the validaterc file.
2. They can _add_ any new custom rules to the validator using a custom rule set. Even if they don't include our spectral rules in their ruleset file, our spectral rules will still be loaded.
3. If a user wants to override our spectral rules using their ruleset file, they would have to:
   - remove that spectral rule from the validaterc file
   - add a rule with the same name to their ruleset file

Reply:
1. Correct
2. Correct
3. Correct, but I realized that the omitting the spectral rule from the validaterc file did _not_ remove it and still overrode the severity the user would specify.

This was due to the config.get() function getting the defaultObject when there was no validaterc found, this default object specified the same rules as the default validaterc and would take priority. So I added a new flag to prevent this defaultObject from overriding it so it acts as you should expect it to like 3.